serde_someip/src/ser.rs

```diff
@@ -36,74 +31,92 @@ impl<'a> ser::Serializer for &'a mut Serializer {
     // into the output string.
 
     fn serialize_bool(self, v: bool) -> Result<()> {
-        Ok(self.output.push(u8::from(v)))
+        self.output.push(u8::from(v));
+        Ok(())
     }
 
     fn serialize_i8(self, v: i8) -> Result<()> {
-        Ok(self.output.push(v as u8))
+        self.output.push(v as u8);
+        Ok(())
     }
 
     fn serialize_i16(self, v: i16) -> Result<()> {
-        Ok(self.output.extend(v.to_ne_bytes()))
+        self.output.extend(v.to_ne_bytes());
+        Ok(())
     }
 
     fn serialize_i32(self, v: i32) -> Result<()> {
-        Ok(self.output.extend(v.to_ne_bytes()))
+        self.output.extend(v.to_ne_bytes());
+        Ok(())
     }
 
-    fn serialize_i64(self, v: i64) -> Result<()>{
+    fn serialize_i64(self, _v: i64) -> Result<()> {
         Err(Error::Syntax)
     }
 
     fn serialize_u8(self, v: u8) -> Result<()> {
-        Ok(self.output.push(v))
+        self.output.extend(v.to_ne_bytes());
+        Ok(())
     }
 
     fn serialize_u16(self, v: u16) -> Result<()> {
-        Ok(self.output.extend(v.to_ne_bytes()))
+        self.output.extend(v.to_ne_bytes());
+        Ok(())
     }
 
     fn serialize_u32(self, v: u32) -> Result<()> {
-        Ok(self.output.extend(v.to_ne_bytes()))
+        self.output.extend(v.to_ne_bytes());
+         Ok(())
     }
 
-    fn serialize_u64(self, v: u64) -> Result<()> {
+    fn serialize_u64(self, _v: u64) -> Result<()> {
         Err(Error::Syntax)
     }
 
     fn serialize_f32(self, v: f32) -> Result<()> {
-        Ok(self.output.extend(v.to_ne_bytes()))
+        self.output.extend(v.to_ne_bytes());
+        Ok(())
     }
 
     fn serialize_f64(self, v: f64) -> Result<()> {
-        Ok(self.output.extend(v.to_ne_bytes()))
+        self.output.extend(v.to_ne_bytes());
+        Ok(())
     }
 
     fn serialize_char(self, v: char) -> Result<()> {
         let res = v as u8;
-        Ok(self.output.push(res))
+        self.output.push(res);
+         Ok(())
     }
 
     fn serialize_str(self, v: &str) -> Result<()> {
-        Ok(self.output.extend(v.as_bytes()))
+        let len: u32 = (v.len() + 3).try_into().unwrap();
+        self.output.extend(len.to_ne_bytes());
+        self.output.push(239);
```

Comment:
239, 187, 191 : Magic numbers. Please add a comment describing why this is needed. Refer to the SOME/IP specification and UTF8 spec.

Reply:
ok